### PR TITLE
Check overlapping trips

### PR DESCRIPTION
**Pull request by simon-the-sloth**
fixes unicef/etools-issues#86

### CODE REVIEW

File: EquiTrack/t2f/tests/test_overlapping_trips.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+from publics.tests.factories import WBSFactory, BusinessAreaFactory, DSARegionFactory
+from t2f.models import ModeOfTravel, make_travel_reference_number, TravelType
+from t2f.tests.factories import CurrencyFactory
+
+from .factories import TravelFactory
+
+log = logging.getLogger('__name__')
+
+
+class OverlappingTravelsTest(APITenantTestCase):
```
Comment (ewheeler) [inline EquiTrack/t2f/tests/test_overlapping_trips.py]: need additional tests for 
- very close but not overlapping trips (trip 2 begins minutes after trip 1) 
- trips spanning changes in daylight savings
- expected behavior for non-overlapping trips
- editing non-overlapping trips to become overlapping
- `ta_required = False`

File: EquiTrack/t2f/tests/test_overlapping_trips.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+        self.travel.save()
+
+    def test_travel_creation(self):
+        currency = CurrencyFactory()
+        dsa_region = DSARegionFactory()
+
+        data = {'deductions': [],
+                'itinerary': [{'origin': 'Berlin',
+                               'destination': 'Budapest',
+                               'departure_date': '2016-11-15T12:06:55.821490',
```
Comment (ewheeler) [inline EquiTrack/t2f/tests/test_overlapping_trips.py]: dates for itinerary items are not within the trip `start_date` and `end_date`

test should have more realistic data

File: EquiTrack/t2f/serializers/travel.py [hunk 3]
```diff
@@ -291,6 +293,23 @@ def validate_itinerary(self, value):
...
+            traveler = attrs.get('traveler', None)
+            if not traveler and self.instance:
+                traveler = self.instance.traveler
+
+            start_date = attrs.get('start_date', getattr(self.instance, 'start_date', None))
+            end_date = attrs.get('end_date', getattr(self.instance, 'end_date', None))
+
+            if start_date and end_date:
+                travel_q = Q(traveler=traveler)
+                travel_q &= ~Q(status__in=[Travel.PLANNED, Travel.CANCELLED])
```
Comment (ewheeler) [inline EquiTrack/t2f/serializers/travel.py]: pls add comments documenting expected behavior of these queries

File: EquiTrack/t2f/tests/test_overlapping_trips.py (hunk 1 above)
Comment (simon-the-sloth) [inline EquiTrack/t2f/tests/test_overlapping_trips.py]: Added all but `ta_required`. We don't have the flow for that yet.

File: EquiTrack/t2f/tests/test_overlapping_trips.py [hunk 4]
```diff
@@ -0,0 +1,255 @@
...
+        response = self.forced_auth_req('patch', reverse('t2f:travels:details:state_change',
+                                                        kwargs={'travel_pk': response_json['id'],
+                                                                'transition_name': 'submit_for_approval'}),
+                                        data=response_json, user=self.unicef_staff)
+        response_json = json.loads(response.rendered_content)
+        self.assertEqual(response_json,
+                         {'non_field_errors': ['You have an existing trip with overlapping dates. '
+                                               'Please adjust your trip accordingly.']})
+
+    def test_daylight_saving(self):
```
Comment (ntrncic) [inline EquiTrack/t2f/tests/test_overlapping_trips.py]: could you add some comments explaining what this test does

**Pull request merged**